src/eventgrid/azext_eventgrid/custom.py

```diff
@@ -275,21 +275,22 @@ def cli_event_subscription_list(   # pylint: disable=too-many-return-statements
         topic_type_name=None):
     if source_resource_id is not None:
         # If Source Resource ID is specified, we need to list event subscriptions for that particular resource.
+        # Since a full resource ID is specified, it should override all other defaults such as default location and RG
         # No other parameters must be specified
-        if (resource_group_name is not None or location is not None or
-                topic_type_name is not None or resource_id is not None):
-            raise CLIError('usage error: Since --source-resource-id is specified, none of the other parameters must'
+        if (topic_type_name is not None or resource_id is not None):
+            raise CLIError('usage error: Since --source-resource-id is specified, none of the other parameters must '
                            'be specified.')
 
         return _list_event_subscriptions_by_resource_id(client, source_resource_id)
 
     if resource_id is not None:
         # DEPRECATED
         # If resource ID is specified, we need to list event subscriptions for that particular resource.
+        # Since a full resource ID is specified, it should override all other defaults such as default location and RG
         # No other parameters must be specified
-        if resource_group_name is not None or location is not None or topic_type_name is not None:
-            raise CLIError('usage error: Since --resource-id is specified, none of the other parameters must'
-                           ' be specified.')
+        if topic_type_name is not None:
+            raise CLIError('usage error: Since --resource-id is specified, none of the other parameters must '
+                           'be specified.')
```

Comment:
If you still want to check for a _user-inputted_ resource-group/location, you can do so still through a validator:
ex: https://github.com/Azure/azure-cli/blob/dev/src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py#L936

Up to you.

Reply:
Thanks @williexu for the info, since we have to release this week, I will skip this for now and will add it in the future. I didn't fully understand the approach though: I see that getattr() is a built-in Python function, and I should check for the 'is_default' property. In the case of resource-group/location, what object should I check for (i.e. the first argument to the getattr() call)?

Reply:
Gotcha. You would check `namespace.resource_group_name`. The namespace is available through a validator.